### ISSUE TITLE
A sequence of operations results in widly incorrect result

### ISSUE THREAD

**Issue by cskr** [paupino/rust-decimal#314]
The following code prints a wildly incorrect value.

```
use rust_decimal_macros::dec;

fn main() {
    let x = dec!(5000) / dec!(1000.26957490549);
    let y = x * dec!(1228.87000756);
    println!("{}", y / dec!(2) / dec!(1228.87000756));
}
```

**Expected Result:** 2.499326244
**Actual Result:** 0.0000000028395229317693263811
**Rust Version:** 1.49.0
**rust_decimal Version:** 1.10.1
**OS:** Debian Unstable

**Comment by paupino** [comment]
Thank you for this write up. I've managed to reproduce it based on the final division in the sequence. I'll look at getting a fix out for this once I track down the issue, however if you need a faster fix you can enable the feature `legacy-ops` which gives a more accurate result.

**Comment by paupino** [comment]
Ok, good news is I have uncovered the issue. I'll run a couple more tests and then publish a fix.

**Comment by paupino** [comment]
Published 1.10.2 which should fix this issue for you. At the moment I have a few [generated tests](https://github.com/paupino/rust-decimal/tree/master/tests/generated) however I may change this from 10000 each sequence to 100000.
